collections/codec/time.go

```diff
@@ -0,0 +1,54 @@
+package codec
+
+import (
+	"encoding/binary"
+	"fmt"
+	"time"
+)
+
+var timeSize = 8
+
+type timeKey[T time.Time] struct{}
+
+func NewTimeKey[T time.Time]() KeyCodec[T] { return timeKey[T]{} }
```

Comment:
if it's time, the time parameter is not needed here

Reply:
done!